src/main/java/com/cloudbees/jenkins/support/timer/DeadlockTrackChecker.java

```diff
@@ -58,7 +55,8 @@ protected void doRun() throws Exception {
 
                 for (ThreadInfo threadInfo : deadLockThreads) {
                     try {
-                        ThreadDumps.printThreadInfo(builder, threadInfo, mbean, filter);
+                        ThreadDumps.printThreadInfo(builder, threadInfo, mbean, 
```

Comment:
I guess if we get the filter before the for loop, it will still be valid? And so we avoid loading it per each thread.

Reply:
good point

Reply:
Not a blocker, but you could have added the orElse() at the beginning as well.